### ISSUE TITLE
0.4.13 on OSX

### ISSUE THREAD

**Issue by breichholf** [Cibiv/NextGenMap#9]
Compiled with XCode 7.1 (AppleClang 7.0.0.7000176) which initially works fine. There are some minor compiler warnings in bamtools and seqan, but especially the former should be fixed with v2.4, according to their repo. In fact, I just tried it and bamtools-v2.4.0 does fix compiler warnings and compiles fine.

Running ngm then, however it crashes fairly early on:
When using the `--output` parameter it crashes with the message:
```
[NGM] Opening for output (SAM): �]�
[FILTER] Unable to open output file �]�
This error is fatal. Quitting...
```
When redirecting to stdout it crashes later on, as it thinks there are 0 reads in the input file.
I think the error arises somewhere entirely different: `Config.GetString` or `InternalGet` seems to be the culprit. When debugging I do get proper return value, but couldn't track it down what the problem then seems to be. I did change the input and output names in the command line, but that only made it worse:
```
[MAIN] Query file (�_�) does not exist.
```
Maybe a problem with iostream/sstream?

**Comment by philres** [comment]
I have encountered the same problem.

A temporary fix is to install gcc from macports and compile NextGenMap with gcc instead of clang.

Eg. after installing gcc 4.9 from macports:

cmake -D CMAKE_C_COMPILER=gcc-mp-4.9 -D CMAKE_CXX_COMPILER=g++-mp-4.9 ..
make

I'll look into the issue with clang.

**Comment by philres** [comment]
Ok I found the problem.

InternalGet returns a string from a map. GetString takes that string, calls c_str() and returns the char *.
Since the string was returned by value from InternalGet, it was deleted after GetString returns. Thus using the returned char * causes a problem.

I fixed it by making InternalGet return the string by reference. Than GetString gets the string from the map which exists also after GetString returned.

Thanks for the bug report and for pointing me into the right direction!

**Comment by philres** [comment]
Did this fix the problem for you as well?

**Comment by breichholf** [comment]
Sorry for the late reply. :+1: compiles and runs fine (albeit much slower than on the linux cluster -- even with 1 core) on my MacPro workstation, thanks!

**Comment by breichholf** [comment]
Probably also can close and fix #6 with this PR.

**Comment by philres** [comment]
That's interesting. I'll have a look at it. 
I think it would be best to open a new issue for that, so I'll close this one.
